Add small script to setup label in lesson repo

rgaiacs opened the pull request:
@gvwilson Do you approve this?

@willingc If @gvwilson approve this, could you review it?

Commit:
Add small script to setup label in lesson repo

The script make easy to create the labels and keep it compatible
across repositories.

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#
+# This script setup the labels of your GitHub repository.
```
This script sets up labels for issues in your GitHub repository.

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+#
+# Syntax:
+#
+#     $ tools/setup-label owner repo
```
Maybe add an example as well (like the docs for `tools/create`)?

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+
+owner=$1
+repo=$2
+labels_and_colors="{\"name\":\"getting-started\",\"color\":\"fbca04\"} {\"name\":\"work-in-progress\",\"color\":\"f7c6c7\"} {\"name\":\"discussion\",\"color\":\"5319e7\"}"
```
By convention, variables in the shell (and shell scripts) are UPPER_CASED - see the repo creation script.

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+labels_and_colors="{\"name\":\"getting-started\",\"color\":\"fbca04\"} {\"name\":\"work-in-progress\",\"color\":\"f7c6c7\"} {\"name\":\"discussion\",\"color\":\"5319e7\"}"
+
+echo "Before setup the labels for http://github.com/${owner}/${repo}"
+echo "you must provide some informations."
```
Aren't you getting the owner's name on the command line? Usage on line 7 suggests you are...

rgaiacs replied:
Yes. There are some use cases that the user will be different from the owner, e.g., https://github.com/swcarpentry/bc.

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+    curl -X POST \
+        -u ${username}:${password} \
+        -d "${tuple}" \
+        "https://api.github.com/repos/${owner}/${repo}/labels"
```
Check return code and exit with meaningful message if failure.

rgaiacs replied:
GitHub will return a JSON that is meaningful for me but maybe not for novice.

In case of failure user will get something like

```
{
  "message": "Validation Failed",
  "documentation_url": "https://developer.github.com/v3/issues/labels/#create-a-label",
  "errors": [
    {
      "resource": "Label",
      "code": "invalid",
      "field": "color"
    }
  ]
}
```

and in case of success

```
{
  "url": "https://api.github.com/repos/r-gaia-cs/bc/labels/getting-started",
  "name": "getting-started",
  "color": "fbca04"
}
```

gvwilson replied:
OK, so we detect that return and print a failure message without showing 
the JSON.

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,26 @@
+read username
+echo "Your GitHub password:"
+read -s password
+
```
Check that repository exists using a HEAD before trying to operate on it (see example in workshop creation script).

rgaiacs commented:
Address @gvwilson comments at 94938a5 (see the commit message for details).

Commit:
Improve script to setup labels

Address @gvwilson comments:

- add an example
- variables in the shell are UPPER_CASED
- Check that repository exists using a HEAD

Commit:
Avoid JSON outup from setup-labels

As requested by @gvwilson, don't print the JSON response from
GitHub.

Others changes:

- Use arrays to store the label and its color
- Print message if fail

gvwilson commented on tools/setup-labels:
```diff
@@ -0,0 +1,67 @@
+then
+    echo "Missing parameters."
+    echo
+    grep '^##' tools/setup-labels | sed 's/## //' | sed 's/##//'
```
Why not just sed 's/##//g' ? or sed -n 's/^## //p' ?

rgaiacs replied:
`sed 's/##//g'` will leave the lines begin with white-space and `sed 's/^## //g'` will not match some lines since they don't end with white-space. I tried `sed 's/^##?//'` but it didn't work.

gvwilson replied:
I think we want to leave the lines with whitespace, don't we? To space things out and make them more readable?
